server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -302,16 +300,13 @@ public ClusterState execute(ClusterState currentState) throws Exception {
                     AliasValidator.validateAlias(alias, request.index(), currentState.metadata());
                 }
 
-                // we only find a template when its an API call (a new index)
-                // find templates, highest order are better matching
-                List<IndexTemplateMetadata> templates =
-                        MetadataIndexTemplateService.findTemplates(currentState.metadata(), request.index());
-
-
-                Map<String, AliasMetadata> templatesAliases = new HashMap<>();
-
-                List<String> templateNames = new ArrayList<>();
-
+                // No template handling here since neither of usages of this service is relevant to them:
+                // 1. Create non-partitioned tables, templates are not related. Partitioned tables are handled by MetadataIndexTemplateService.
+                // 2. Resize partitioned tables. findTemplates always returns an empty list since
+                //    request.index() has prefix "resized" which doesn't match table pattern. See testShrinkShardsOfPartition
+                // 3. Creating partitions/indices by inserting into a partitioned table.
+                //    We don't use this service anymore after introducing https://github.com/crate/crate/commit/f1c96b517d6d4f31ada5c7b42da49f5a41c12869
+                //    where we have dedicated TransportCreatePartitionsAction, which is an optimized version of MetadataCreateIndexService
```

Comment:
Could we add a `assert hasNoMatchingTemplate(...)` as a defensive sanity check and to ensure these assumptions aren't invalidated in the future?

Reply:
I'd go with an assert because it communicates a different intent. This case can't happen unless we messed up or if we change things in the future.

Reply:
Some tests are failing, basically with this assertion: will have another look. Thanks for pointing it out!

Reply:
ok, tests are failing because assertion uses `PartitionName.templateName` which already contains validation. 
I removed validation.

Reply:
I restored validation and added a flag to skip it. There are some more validations failing because of unexpected usage.
The fact they fail is good but will add some more skip validation logic to be able to assert without failures.

Reply:
ended up wrapping it with a try block - validation always fails on resizing partitioned tables but this is DDL so not. big deal performance wise  (stack trace generation). 

Otherwise I need either not to do assertion at all or introduce some skipValidation flag in PartitionName methods and in the IndexParts which itself also a bit of mess.